src/main/java/lab5/ui/Console.java

```diff
@@ -1,116 +1,14 @@
-     */
-    public static void printAlert(String message) {
-        System.out.println(fAlert.format(message));
-    }
-
-    /**
-     * Repeatedly prompt the user to select a player for the given token
-     * until the select a valid one of a set of valid players
-     * @param whichPlayer The player for which to prompt
-     * @return A player object representing the user's chosen player
-     */
-    public static Player promptForPlayer(PlayerToken whichPlayer) {
-
-        while ( true ) {
-            var input = prompt(fPrompt.format("Who will play " + whichPlayer + "? "));
-
-            if ( input.startsWith("@") ) {
-                input = input.substring(1).toLowerCase(); // remove the '@' prefix
-
-                switch ( input ) {
-                    // E.g.
-                    // case "randy" -> { return new Randy(); }
-                    default -> printAlert("TODO: Implement computer players");
-                }
-            } else {
-                return new Player(input);
-            }
-        }
-    }
-
-    /**
-     * Display the given game board
-     * @param board A tictactoe game board
-     */
-    public static void showBoard(Board board) {
-        var sb = new StringBuilder();
-        for (var c : board.toString().toCharArray()) {
-            if ( c == 'X' ) {
-                sb.append(colorize("X", BRIGHT_CYAN_TEXT()));
-            } else if ( c == 'O' ) {
-                sb.append(colorize("O", BRIGHT_MAGENTA_TEXT()));
-            } else {
-                sb.append(c);
-            }
-        }
-        System.out.println(sb);
-    }
-
-    /**
-     * Repeatedly prompt the user for a position on which to place their next token.
-     * If they enter an invalid response they are re-prompted.
-     * @param prompt The prompt to display to the user
-     * @return The position selected by the user
-     */
-    public static Position promptForPosition(String prompt) {
-
-        final String helpMessage = "Input must be in the format 'row column', e.g., '1 2' or 't m' for the top middle cell.";
-
-        while ( true ) {
-            var input = prompt(fPrompt.format(prompt));
-
-            if ( input.length() != 3 ) {
-                printAlert(helpMessage);
-                continue;
-            }
-
-            var parts = input.split(" ");
-
-            if ( parts.length != 2 ) {
-                printAlert(helpMessage);
-                continue;
-            }
-
-            try {
-                return new Position(Row.from(parts[0]), Col.from(parts[1]));
-            } catch ( IllegalArgumentException e ) {
-                printAlert(helpMessage);
-            }
-        }
+    public Player promptForPlayer(PlayerToken token) {
+        // Prompt the user for a player name and create a Player object
+        System.out.println("Enter name for player " + token + ":");
+        String name = System.console().readLine();
+        return new Player(name, token);
```

Comment:
This doesn't work because Player is an abstract class and therefore cannot be instantiated.  See the sample solution.